todo_app/templates/index_read.html

```diff
@@ -0,0 +1,130 @@
+{% extends "layout.html" %}
```

Comment:
Nothing wrong with this approach, but it looks like a fair amount of duplication - might be worth considering whether it would be easier (even if not worth switching) to pass the role through as part of the view model, and conditionally render certain sections